File: BeachVolleyball/Team.cs
//---------------------------------------------------------------------------------------------------------------------
// <copyright file="Team.cs" company="Microsoft">
//     Copyright (c) Microsoft. All rights reserved.
// </copyright>
//---------------------------------------------------------------------------------------------------------------------

namespace BeachVolleyball
{
    public class Team
    {
        public Team(Player player1, Player player2)
        {
            this.Player1 = player1;
            this.Player2 = player2;
        }

        public Player Player1 { get; }

        public Player Player2 { get; }

        public double Rank => this.Player1.Rank + this.Player2.Rank;

        public double PreviousYearRank => this.Player1.PreviousYearRank + this.Player2.PreviousYearRank;

        public override string ToString()
        {
            return string.Format("({0}, {1} \t2019 Rank: {2}, 2018 Rank: {3})", this.Player1, this.Player2, this.Rank, this.PreviousYearRank);
        }
    }
}

File: BeachVolleyball.Server/ClientApp/src/components/Tournaments/Team.js
import React from "react";
import styled from "styled-components";
import Player from "./Player";

const Team = ({ player1, player2, rank, previousYearRank }) => (
  <TeamContainer>
    <Player1 {...player1} />
    <Player2 {...player2} />
    <Rank>{rank}</Rank>
  </TeamContainer>
);

const TeamContainer = styled.div`
  display: grid;
  grid-template-rows: repeat(2, auto);
  grid-template-columns: 30px auto;
  justify-items: start;

  & > * {
      margin-bottom: 10px;
  }
`;
const Player1 = styled(Player)`
  grid-row: 1/2;
  grid-column: 1/2;
`;

const Player2 = styled(Player)`
  grid-row: 2/3;
  grid-column: 1/2;
`;

const Rank = styled.p`
  grid-row: 1/3;
  grid-column: 1/2;

  align-self: center;
`;

export default Team;
